src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataDeltaResourceSetWrapper.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>
+    public sealed class ODataDeltaResourceSetWrapper : ODataResourceSetWrapperBase
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataResourceSetWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataDeltaResourceSetWrapper(ODataResourceSetBase item)
+            : base(item)
+        {
+            ResourceSetType = ResourceSetType.DeltaResourceSet;
```

Comment:
>ResourceSetType = ResourceSetType.DeltaResourceSet; [](start = 12, length = 51)

you can just return it as;


internal override ResourceSetType ResourceSetType => ResourceSetType.DeltaResourceSet;

Reply:
Would it be better to set it on the base type in case the base type tries to do anything with the ResourceSetType?

---
In reply to: [512292387](https://github.com/OData/WebApi/pull/2275#discussion_r512292387) [](ancestors = 512292387)